src/StructTypes.jl

```diff
@@ -9,8 +9,15 @@ StructType(x::T) where {T} = StructType(T)
 
 "Default `StructTypes.StructType` for types that don't have a `StructType` defined; this ensures objects must have an explicit `StructType` to avoid unanticipated issues"
 struct NoStructType <: StructType end
+struct SingletonType <: StructType end
```

Comment:
I also wonder if we should make this:
```julia
struct SingletonType <: Struct end
```
so it's less breaking for downstream packages.

But maybe not, since they'll need to update to support it anyway. If we change the default struct type to `Struct`, that would force at least a minor version bump anyway, so packages could update their bounds and properly support `SingletonType` no matter what it subtypes, so this is probably fine.